File: FSM/Entity/QuoteContractOriginals/QuoteContractOriginalSQL.cs
// Decompiled with JetBrains decompiler
// Type: FSM.Entity.QuoteContractOriginals.QuoteContractOriginalSQL
// Assembly: FSM, Version=1.12.1.0, Culture=neutral, PublicKeyToken=null
// MVID: FAB6F0E2-B68A-47AB-AF95-A9E31DED7575
// Assembly location: C:\github\gabriel.fieldservicemanager\src\bin\x86\Debug\FSM.exe

using FSM.Entity.Sys;
using Microsoft.VisualBasic.CompilerServices;
using System.Data;
using System.Runtime.CompilerServices;

namespace FSM.Entity.QuoteContractOriginals
{
  public class QuoteContractOriginalSQL
  {
    private Database _database;

    public QuoteContractOriginalSQL(Database database)
    {
      this._database = database;
    }

    public DataTable VisitFrequency_Get()
    {
      this._database.ClearParameter();
      return this._database.ExecuteSP_DataTable(nameof (VisitFrequency_Get), true);
    }

    public QuoteContractOriginal Get(int QuoteContractID)
    {
      this._database.ClearParameter();
      this._database.AddParameter("@QuoteContractID", (object) QuoteContractID, false);
      DataTable dataTable = this._database.ExecuteSP_DataTable("QuoteContractOriginal_Get", true);
      if (dataTable == null || dataTable.Rows.Count <= 0)
        return (QuoteContractOriginal) null;
      QuoteContractOriginal contractOriginal1 = new QuoteContractOriginal();
      QuoteContractOriginal contractOriginal2 = contractOriginal1;
      contractOriginal2.IgnoreExceptionsOnSetMethods = true;
      contractOriginal2.SetQuoteContractID = RuntimeHelpers.GetObjectValue(dataTable.Rows[0][nameof (QuoteContractID)]);
      contractOriginal2.SetCustomerID = RuntimeHelpers.GetObjectValue(dataTable.Rows[0]["CustomerID"]);
      contractOriginal2.SetQuoteContractReference = RuntimeHelpers.GetObjectValue(dataTable.Rows[0]["QuoteContractReference"]);
      contractOriginal2.QuoteContractDate = Conversions.ToDate(dataTable.Rows[0]["QuoteContractDate"]);
      contractOriginal2.ContractStart = Conversions.ToDate(dataTable.Rows[0]["ContractStart"]);
      contractOriginal2.ContractEnd = Conversions.ToDate(dataTable.Rows[0]["ContractEnd"]);
      contractOriginal2.SetQuoteContractStatusID = RuntimeHelpers.GetObjectValue(dataTable.Rows[0]["QuoteContractStatusID"]);
      contractOriginal2.SetQuoteContractPrice = RuntimeHelpers.GetObjectValue(dataTable.Rows[0]["QuoteContractPrice"]);
      contractOriginal2.SetReasonForReject = RuntimeHelpers.GetObjectValue(dataTable.Rows[0]["ReasonForReject"]);
      contractOriginal2.SetDeleted = Conversions.ToBoolean(dataTable.Rows[0]["Deleted"]);
      contractOriginal1.Exists = true;
      return contractOriginal1;
    }

    public double QuoteContractCalculatedTotal(int QuoteContractID)
    {
      this._database.ClearParameter();
      this._database.AddParameter("@QuoteContractID", (object) QuoteContractID, true);
      return Helper.MakeDoubleValid(RuntimeHelpers.GetObjectValue(this._database.ExecuteSP_OBJECT("QuoteContractOriginalCalculatedTotal", true)));
    }

    public QuoteContractOriginal Insert(QuoteContractOriginal oQuoteContract)
    {
      this._database.ClearParameter();
      this.AddQuoteContractParametersToCommand(ref oQuoteContract);
      oQuoteContract.SetQuoteContractID = (object) Helper.MakeIntegerValid(RuntimeHelpers.GetObjectValue(this._database.ExecuteSP_OBJECT("QuoteContractOriginal_Insert", true)));
      oQuoteContract.Exists = true;
      return oQuoteContract;
    }

    public void Update(QuoteContractOriginal oQuoteContract)
    {
      this._database.ClearParameter();
      this._database.AddParameter("@QuoteContractID", (object) oQuoteContract.QuoteContractID, true);
      this.AddQuoteContractParametersToCommand(ref oQuoteContract);
      this._database.ExecuteSP_NO_Return("QuoteContractOriginal_Update", true);
    }

    public void Delete(int QuoteContractID)
    {
      this._database.ClearParameter();
      this._database.AddParameter("@QuoteContractID", (object) QuoteContractID, true);
      this._database.ExecuteSP_NO_Return("QuoteContractOriginal_Delete", true);
    }

    private void AddQuoteContractParametersToCommand(ref QuoteContractOriginal oQuoteContract)
    {
      Database database = this._database;
      database.AddParameter("@CustomerID", (object) oQuoteContract.CustomerID, true);
      database.AddParameter("@QuoteContractReference", (object) oQuoteContract.QuoteContractReference, true);
      database.AddParameter("@QuoteContractDate", (object) oQuoteContract.QuoteContractDate, true);
      database.AddParameter("@ContractStart", (object) oQuoteContract.ContractStart, true);
      database.AddParameter("@ContractEnd", (object) oQuoteContract.ContractEnd, true);
      database.AddParameter("@QuoteContractStatusID", (object) oQuoteContract.QuoteContractStatusID, true);
      database.AddParameter("@QuoteContractPrice", (object) oQuoteContract.QuoteContractPrice, true);
      database.AddParameter("@ReasonForReject", (object) oQuoteContract.ReasonForReject, true);
    }
  }
}

File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOriginal/QuoteContractOriginal/QuoteContractOriginalSQL.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace QuoteContractOriginals

        Public Class QuoteContractOriginalSQL
            Private _database As Sys.Database

            Public Sub New(ByVal database As Sys.Database)
                _database = database
            End Sub

#Region "Functions"

            Public Function VisitFrequency_Get() As DataTable
                _database.ClearParameter()
                Return _database.ExecuteSP_DataTable("VisitFrequency_Get")
            End Function

            Public Function [Get](ByVal QuoteContractID As Integer) As QuoteContractOriginal
                _database.ClearParameter()
                _database.AddParameter("@QuoteContractID", QuoteContractID)

                'Get the datatable from the database store in dt
                Dim dt As DataTable = _database.ExecuteSP_DataTable("QuoteContractOriginal_Get")

                If Not dt Is Nothing Then
                    If dt.Rows.Count > 0 Then
                        Dim oQuoteContract As New QuoteContractOriginal
                        With oQuoteContract
                            .IgnoreExceptionsOnSetMethods = True
                            .SetQuoteContractID = dt.Rows(0).Item("QuoteContractID")
                            .SetCustomerID = dt.Rows(0).Item("CustomerID")
                            .SetQuoteContractReference = dt.Rows(0).Item("QuoteContractReference")
                            .QuoteContractDate = CDate(dt.Rows(0).Item("QuoteContractDate"))
                            .ContractStart = CDate(dt.Rows(0).Item("ContractStart"))
                            .ContractEnd = CDate(dt.Rows(0).Item("ContractEnd"))
                            .SetQuoteContractStatusID = dt.Rows(0).Item("QuoteContractStatusID")
                            .SetQuoteContractPrice = dt.Rows(0).Item("QuoteContractPrice")
                            .SetReasonForReject = dt.Rows(0).Item("ReasonForReject")
                            .SetDeleted = dt.Rows(0).Item("Deleted")
                        End With
                        oQuoteContract.Exists = True
                        Return oQuoteContract
                    Else
                        Return Nothing
                    End If
                Else
                    Return Nothing
                End If
            End Function

            Public Function QuoteContractCalculatedTotal(ByVal QuoteContractID As Integer) As Double
                _database.ClearParameter()
                _database.AddParameter("@QuoteContractID", QuoteContractID, True)

                Return Entity.Sys.Helper.MakeDoubleValid(_database.ExecuteSP_OBJECT("QuoteContractOriginalCalculatedTotal"))
            End Function

            Public Function Insert(ByVal oQuoteContract As QuoteContractOriginal) As QuoteContractOriginal
                _database.ClearParameter()
                AddQuoteContractParametersToCommand(oQuoteContract)

                oQuoteContract.SetQuoteContractID = Entity.Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("QuoteContractOriginal_Insert"))
                oQuoteContract.Exists = True
                Return oQuoteContract
            End Function

            Public Sub Update(ByVal oQuoteContract As QuoteContractOriginal)
                _database.ClearParameter()
                _database.AddParameter("@QuoteContractID", oQuoteContract.QuoteContractID, True)
                AddQuoteContractParametersToCommand(oQuoteContract)
                _database.ExecuteSP_NO_Return("QuoteContractOriginal_Update")
            End Sub

            Public Sub Delete(ByVal QuoteContractID As Integer)
                _database.ClearParameter()
                _database.AddParameter("@QuoteContractID", QuoteContractID, True)
                _database.ExecuteSP_NO_Return("QuoteContractOriginal_Delete")
            End Sub

            Private Sub AddQuoteContractParametersToCommand(ByRef oQuoteContract As QuoteContractOriginal)
                With _database
                    .AddParameter("@CustomerID", oQuoteContract.CustomerID, True)
                    .AddParameter("@QuoteContractReference", oQuoteContract.QuoteContractReference, True)
                    .AddParameter("@QuoteContractDate", oQuoteContract.QuoteContractDate, True)
                    .AddParameter("@ContractStart", oQuoteContract.ContractStart, True)
                    .AddParameter("@ContractEnd", oQuoteContract.ContractEnd, True)
                    .AddParameter("@QuoteContractStatusID", oQuoteContract.QuoteContractStatusID, True)
                    .AddParameter("@QuoteContractPrice", oQuoteContract.QuoteContractPrice, True)
                    .AddParameter("@ReasonForReject", oQuoteContract.ReasonForReject, True)
                End With
            End Sub

#End Region

        End Class

    End Namespace

End Namespace


